crates/compiler/tests/language_tests.rs

```diff
@@ -0,0 +1,120 @@
+//! Adapted from <https://github.com/YarnSpinnerTool/YarnSpinner/blob/da39c7195107d8211f21c263e4084f773b84eaff/YarnSpinner.Tests/LanguageTests.cshttps://github.com/YarnSpinnerTool/YarnSpinner/blob/da39c7195107d8211f21c263e4084f773b84eaff/YarnSpinner.Tests/LanguageTests.cs>
+
+mod test_base;
+use crate::test_base::*;
+
+#[test]
+#[ignore]
+fn test_example_script() {
+    todo!()
+}
+
+#[test]
+#[ignore]
+fn test_merging_nodes() {
+    todo!()
+}
+
+#[test]
+#[ignore]
+fn test_end_of_notes_with_options_not_added() {
+    todo!()
+}
+
+#[test]
+fn test_node_headers() {
+    let path = TestBase::test_data_path().join("Headers.yarn");
+    let result = Compiler.Compile(CompilationJob.CreateFromFiles(path));
+
+    result.Diagnostics.Should().BeEmpty();
+
+    result.Program.Nodes.Count.Should().Be(6);
+
+    foreach (let tag in new[] {"one", "two", "three"})
+    {
+        result.Program.Nodes["Tags"].Tags.Should().Contain(tag);
+    }
+
+    let headers = new Dictionary<string, List<(string, string)>>();
+    headers.Add("EmptyTags", new List<(string, string)>{
+        ("title","EmptyTags"),
+        ("tags", null)
+    });
+    headers.Add("ArbitraryHeaderWithValue", new List<(string, string)>{
+        ("title", "ArbitraryHeaderWithValue"),
+        ("arbitraryheader", "some-arbitrary-text")
+    });
+    headers.Add("Tags", new List<(string, string)>{
+        ("title", "Tags"),("tags",
+                           "one two three")
+    });
+    headers.Add("SingleTagOnly", new List<(string, string)>{
+        ("title", "SingleTagOnly")
+    });
+    headers.Add("Comments", new List<(string, string)>{
+        ("title", "Comments"),
+        ("tags", "one two three")
+    });
+    headers.Add("LotsOfHeaders", new List<(string, string)>{
+        ("contains", "lots"),
+        ("title", "LotsOfHeaders"),
+        ("this", "node"),
+        ("of", null),
+        ("blank", null),
+        ("others", "are"),
+        ("headers", ""),
+        ("some", "are"),
+        ("not", "")
+    });
+
+    headers.Count.Should().Be(result.Program.Nodes.Count);
+    foreach (let pair in headers)
+    {
+        result.Program.Nodes[pair.Key].Headers.Count.Should().Be(pair.Value.Count);
+
+        // go through each item in the headers and ensure they are in the header list
```

Comment:
this comment doesn't add value

Reply:
Agreed, it's copy-pasted from the original implementation. I'll remove it when I finish rustifying this test